File: MiniDOMAsteriskLib/CallManagers/Events/Alarm.vb
Imports System.Net.Sockets
Imports System.Threading
Imports System.Net

Namespace CallManagers.Events

    
    Public Class Alarm
        Inherits AsteriskEvent

        Public Sub New()
            MyBase.New("Alarm")
        End Sub

        Public Sub New(ByVal rows() As String)
            MyBase.New(rows)
        End Sub

        ''' <summary>
        ''' (Red|Yellow|Blue|No|Unknown) Alarm|Recovering|Loopback|Not Open|None
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public ReadOnly Property Alarm As String
            Get
                Return Me.GetAttribute("Alarm")
            End Get
        End Property

        Public ReadOnly Property Channel As String
            Get
                Return Me.GetAttribute("Channel")
            End Get
        End Property
         
    End Class

End Namespace

File: MiniDOMAsteriskLib/CallManagers/Events/Alarm.cs

namespace minidom.CallManagers.Events
{
    public class AlarmType : AsteriskEvent
    {
        public AlarmType() : base("Alarm")
        {
        }

        public AlarmType(string[] rows) : base(rows)
        {
        }

        /// <summary>
        /// (Red|Yellow|Blue|No|Unknown) Alarm|Recovering|Loopback|Not Open|None
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        public string Alarm
        {
            get
            {
                return GetAttribute("Alarm");
            }
        }

        public string Channel
        {
            get
            {
                return GetAttribute("Channel");
            }
        }
    }
}